apps/astarte_housekeeping/lib/astarte_housekeeping/engine.ex

```diff
@@ -161,6 +162,48 @@ defmodule Astarte.Housekeeping.Engine do
     end
   end
 
+  defp update_device_registration_limit(realm_name, new_limit) do
+    _ = Logger.info("Updating device registration limit", tag: "update_device_registration_limit")
+
+    case new_limit do
+      nil -> :ok
+      :remove_limit -> delete_device_registration_limit(realm_name)
+      n when is_integer(n) -> set_device_registration_limit(realm_name, n)
+    end
+  end
```

Comment:
You're logging even if you don't actually update the limit. Pattern match in the function head and map directly to the noop/delete/set implementations, so you can also have a more precise log ("Removing/Updating device registration limit")